reactor-netty-core/src/test/java/reactor/netty/resources/PooledConnectionProviderCustomMetricsTest.java

```diff
@@ -108,16 +120,46 @@ private void triggerAcquisition(boolean metricsEnabled, Supplier<ConnectionProvi
 		ClientTransportConfigImpl config =
 				new ClientTransportConfigImpl(group, pool, Collections.emptyMap(), remoteAddress);
 
+		Connection conn = null;
 		try {
-			pool.acquire(config, ConnectionObserver.emptyListener(), remoteAddress, config.resolverInternal())
+			conn = pool.acquire(config, ConnectionObserver.emptyListener(), remoteAddress, config.resolverInternal())
 			    .block(Duration.ofSeconds(10L));
 			fail("Exception is expected");
 		}
 		catch (Exception expected) {
 			// ignore
 		}
+		return conn;
 	}
 
+	static final class MeterRegistrarImpl implements ConnectionProvider.MeterRegistrar {
+		AtomicBoolean registered;
+		AtomicBoolean deRegistered;
+		AtomicInteger customMetric;
+
+		MeterRegistrarImpl(AtomicBoolean registered, AtomicBoolean deRegistered, AtomicInteger customMetric) {
+			this.registered = registered;
+			this.deRegistered = deRegistered;
+			this.customMetric = customMetric;
+		}
```

Comment:
Let's announce that `Nullable` is allowed.

```suggestion
		MeterRegistrarImpl(
				@Nullable AtomicBoolean registered,
				@Nullable AtomicBoolean deRegistered,
				@Nullable AtomicInteger customMetric) {
			this.registered = registered;
			this.deRegistered = deRegistered;
			this.customMetric = customMetric;
		}

```